### ISSUE TITLE
arrow-parens: as-needed, not working with async functions

### ISSUE THREAD

**Issue by felixsanz** [babel/babel-eslint#268]
ESLint 2.3.0 using babel-eslint.

On async functions that are also arrow functions and have a single parameter, ESLint doesn't complain about parentheses. Example:

Bad:
```javascript
app.use(async (ctx) => {
// code
})
```

Should be:
```javascript
app.use(async ctx => {
// code
})
```

Basically, "arrow-parens: as-needed" is not affecting async functions

**Comment by kaicataldo** [comment]
Looks like https://github.com/babel/eslint-plugin-babel contains a patched version that has what you're looking for
